log_shipping/clent_log_shipping_s3_bucket_create.go

```diff
@@ -0,0 +1,81 @@
+package log_shipping
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	logzio_client "github.com/logzio/logzio_terraform_client"
+)
+
+const createS3BucketServiceMethod string = http.MethodPost
+const createS3BucketMethodSuccess int = http.StatusOK
+
+// type FieldError struct {
+// 	Field   string
+// 	Message string
+// }
+
+// func (e FieldError) Error() string {
+// 	return fmt.Sprintf("%v: %v", e.Field, e.Message)
+// }
+
+func validateCreateS3BucketRequest(bucket S3Bucket) error {
+
+	if len(bucket.AccessKey) == 0 {
+		return fmt.Errorf("accessKey must be set")
+	}
+
+	// TODO: complete validation
+
+	return nil
+}
+
+func buildCreateS3BucketRequest(bucket S3Bucket) map[string]interface{} {
+	var createBucket = map[string]interface{}{}
+
+	createBucket[fldAwsAccessKey] = bucket.AccessKey
+	createBucket[fldAwsSecretKey] = bucket.SecretKey
+	createBucket[fldAwsArn] = bucket.Arn
+	createBucket[fldAwsBucket] = bucket.Bucket
+	createBucket[fldAwsPrefix] = bucket.Prefix
+	createBucket[fldAwsActive] = bucket.Active
+	createBucket[fldAwsAddS3ObjectKeyAsLogField] = bucket.AddS3ObjectKeyAsLogField
+	createBucket[fldAwsRegion] = bucket.Region
+	createBucket[fldLogsType] = bucket.LogsType
+
+	return createBucket
```

Comment:
Build the object in one line (very similar to this: https://github.com/logzio/docker-logging-plugin/blob/master/driver.go#L367)

Reply:
You can also build a map like in here: https://github.com/logzio/logzio_terraform_client/blob/master/users/client_user_create.go#L29